### PR TITLE
[FLINK-36622] Remove the dependency of StateBenchmark on RocksDBKeyedStateBackend APIs.

### PR DESCRIPTION
**Pull request by AlexYinHan**
## What is the purpose of the change

Currently, flink-benchmarks relies on non-public APIs in Flink. For example, in StateBackendBenchmarkUtils.java, the function compactState takes RocksDBKeyedStateBackend as its first argument.

This requires explicit type conversion in flink-benchmark(from KeyedStateBackend to RocksDBKeyedStateBackend). Moreover, this means that once the signature of RocksDBKeyedStateBackend changes, we need to modify flink-benchmark correspondingly.

Therefore, we should avoid exposing non-public APIs in StateBackendBenchmarkUtils.

## Brief change log

Change the 1st argument of `StateBackendBenchmarkUtils#compactState` from RocksDBKeyedStateBackend to KeyedStateBackend.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-test-utils-parent/flink-test-utils/src/test/java/org/apache/flink/state/benchmark/StateBackendBenchmarkUtils.java [hunk 1]
```diff
@@ -280,10 +280,15 @@ public static <K, S extends State, T> void applyToAllKeys(
...
     }
 
-    public static <K, S extends State, T> void compactState(
-            RocksDBKeyedStateBackend<K> backend, StateDescriptor<S, T> stateDescriptor)
+    public static <K, S extends State, T> boolean compactState(
+            KeyedStateBackend<K> backend, StateDescriptor<S, T> stateDescriptor)
             throws RocksDBException {
-        backend.compactState(stateDescriptor);
+        if (!(backend instanceof RocksDBKeyedStateBackend)) {
+
```
Comment (Zakelly) [inline flink-test-utils-parent/flink-test-utils/src/test/java/org/apache/flink/state/benchmark/StateBackendBenchmarkUtils.java:287]: Redundant line.

File: flink-test-utils-parent/flink-test-utils/src/test/java/org/apache/flink/state/benchmark/StateBackendBenchmarkUtils.java [hunk 2]
```diff
@@ -280,10 +280,15 @@ public static <K, S extends State, T> void applyToAllKeys(
                 function);
     }
 
-    public static <K, S extends State, T> void compactState(
-            RocksDBKeyedStateBackend<K> backend, StateDescriptor<S, T> stateDescriptor)
+    public static <K, S extends State, T> boolean compactState(
```
Comment (Zakelly) [inline flink-test-utils-parent/flink-test-utils/src/test/java/org/apache/flink/state/benchmark/StateBackendBenchmarkUtils.java:283]: I'd suggest a javadoc describing the behavior.

**Review by Zakelly — COMMENTED** [review]
Thanks for the fix! Left two comments

**Comment by AlexYinHan** [comment]
@Zakelly Thanks for the comments. I've resolved them.

**Comment by Zakelly** [comment]
@flinkbot run azure

**Review by Zakelly — APPROVED** [review]

**Comment by Zakelly** [comment]
@flinkbot run azure

**Pull request merged**